javascript.js

```diff
@@ -29,15 +28,15 @@ window.onload = function () {
     var buttons = function () {
         myButtons = document.getElementById("buttons");
         letters = document.createElement("ul");
-    
-        for (var x = 0; x < alphabet.length; x++) {
-            letters.id = "alphabet";
-            list = document.createElement("li");
-            list.id = "letter";
-            list.innerHTML = alphabet[x];
-            clickInput();
-            myButtons.appendChild(letters);
-            letters.appendChild(list);
+        letters.id = "alphabet";
+        myButtons.appendChild(letters);
+        
+        for (var x = 0; x < alphabet.length; x++) {    
+            var letterBtn = document.createElement("li");
+            letterBtn.id = "letter";
+            letterBtn.innerHTML = alphabet[x];
+            letterBtn.onclick = clickInput;    
+            letters.appendChild(letterBtn);
```

Comment:
Renamed "list" for clarity. Moved appending the UL outside of loop - only need to do it once.  Updated usage of clickInput to be an eventHandler - generating it was unnecessary in this case and also there was potential for scope issues.